File: C#/solid/Customer.cs
namespace SOLID;

public sealed class Customer : IUserManagement, ISubscriptionNotificationService
{
    public string Name { get; }
    public string? Email { get; }
    
    public Customer(string name)
    {
        Name = name;
    }

    public void SubscribeToNewProductsAvailability()
        => Console.WriteLine("Subscribing to new products availability...");

    public void SubscribeToSMSNotifications()
        => Console.WriteLine("Subscribing to SMS notifications...");

    public void UpdateUserProfile(Customer customer)
        => Console.WriteLine($"Updating user profile + {customer.Name}...");

    public void ChangePassword(Customer customer, string newPassword)
        => Console.WriteLine($"Updating user: {customer.Name}...");
}

File: kotlin/solid/Customer.kt
package solid

class Customer(val name:String, val email:String): UserManagement{
    override fun updateUserProfile(customer: Customer){
        println("Updating user profile ${customer.name}......" )
    }

    override fun changePassword(customer: Customer, newPassword: String) {
        println("Updating user ${customer.name}.....")
    }


}